### ISSUE TITLE
VSCode extension on MacOS cannot access core tools, can't debug

### ISSUE THREAD

**Issue by RylandDeGregory** [microsoft/vscode-azurefunctions#2746]
When trying to debug a function project locally, VS Code cannot access the installation of azure-functions-core-tools v3.
![Screen Shot 2021-03-12 at 11 21 35 PM](https://user-images.githubusercontent.com/18073815/111018885-c7c0ed80-8389-11eb-9b9a-bec93b3ba198.png)

VS Code also fails to install the tools using the built-in command.
![Screen Shot 2021-03-12 at 11 22 44 PM](https://user-images.githubusercontent.com/18073815/111018904-e921d980-8389-11eb-9f0f-49b4d86056ff.png)

However, azure-functions-core-tools v3 is installed using homebrew and accessible via PATH.
![Screen Shot 2021-03-12 at 11 24 02 PM](https://user-images.githubusercontent.com/18073815/111018942-1a020e80-838a-11eb-9895-74b4be64403e.png)

Is there a VS Code setting or environment variable that I need to set for the extension (and therefore the debugger) to pick up my installation of the core tools?

**Comment by ejizba** [comment]
If it is actually is in your path and we're just blocking debugging for no reason, you can set `"azureFunctions.validateFuncCoreTools": false` in your VS Code settings to bypass our prompt

If that works, I'll close this as a duplicate of https://github.com/microsoft/vscode-azurefunctions/issues/2019. The PATH is sometimes different in the child process we use to detect the `func` cli. The fix on our side would be to use a "task" with the same environment as your VS Code terminal to detect the `func` cli

**Comment by RylandDeGregory** [comment]
I've added the setting, so I can see the error now. However, it still doesn't make sense why the integrated terminal can see the executable but the debug task can't.
<img width="669" alt="Screen Shot 2021-03-15 at 4 11 33 PM" src="https://user-images.githubusercontent.com/18073815/111215172-3af47a80-85a9-11eb-94fd-9e0f150407c2.png">
<img width="396" alt="Screen Shot 2021-03-15 at 4 11 46 PM" src="https://user-images.githubusercontent.com/18073815/111215175-3b8d1100-85a9-11eb-9062-1bc66e94d00d.png">

**Comment by ejizba** [comment]
Yeah I'm not sure why there's a discrepancy. I'd suggest you take a look at your integrated terminal settings: https://code.visualstudio.com/docs/editor/integrated-terminal

Specifically, you might want to make sure the "automation" and normal one are the same:
https://code.visualstudio.com/docs/editor/integrated-terminal#_changing-shell-for-tasks-and-debug

**Comment by ejizba** [comment]
If that's not helpful, please give more information on how you installed the func cli. When you say "VS Code also fails to install the tools using the built-in command." there should be some errors in the "Azure Functions" output pane. You could copy those here and/or try uninstalling through your method first before installing again with the VS Code command

**Comment by RylandDeGregory** [comment]
Here's the settings you referenced from my `settings.json`:
<img width="457" alt="Screen Shot 2021-03-15 at 4 42 09 PM" src="https://user-images.githubusercontent.com/18073815/111218664-65483700-85ad-11eb-9be8-b85ba2df4c06.png">


I installed azure-functions-core-tools@3 using Homebrew based on the instructions here:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=macos%2Ccsharp%2Cbash#v2

I uninstalled and tried to install following your instructions to view the error in the Azure Functions output pane:
<img width="692" alt="Screen Shot 2021-03-15 at 4 38 50 PM" src="https://user-images.githubusercontent.com/18073815/111218416-14384300-85ad-11eb-889e-071c2ce5df37.png">
The link in the error just tells me to install using Homebrew.

**Comment by ejizba** [comment]
Okay so we throw that error if the install command succeeded, but `func --version` still failed. It's probably some underlying issue where brew packages aren't in your path for  VS Code for some reason. Aka, we have no  problem running `brew install azure-functions...`, but we still can't see `func`. Any idea why brew packages  might not be on your PATH? How are you starting VS Code...from a terminal or from the App Icon? What about do you have any .zshrc or .zhenv files that could be affecting this?

**Comment by RylandDeGregory** [comment]
The core tools did not install successfully using the VS Code command, despite that error allegedly indicating that it did.

However, I went and installed the core tools from Homebrew directly `brew install azure-functions-core-tools@3` then followed the instructions here: https://stackoverflow.com/a/45637716 to add my `$PATH` string to my VS Code settings file. 

That solved it. I am now able to debug functions locally from within VS Code (using `F5` or the debug panel).

I'm still not sure why the extension (or VS Code itself) is not picking up my complete `$PATH`, but I'm glad to have found a workaround.

Thank you very much for your assistance @ejizba , and I hope that you are able to find a root cause at some point :)

**Comment by ejizba** [comment]
Glad you got it working! We might not be able to fix the root cause per se (`func` not showing up in the PATH), but we could perhaps make it easier to identify and solve this error for users - maybe in the form of better error messages or more VS Code settings

**Comment by ejizba** [comment]
We added a setting for the path ("azureFunctions.funcCliPath") in https://github.com/microsoft/vscode-azurefunctions/pull/2928, which should help with this scenario. Only thing left is we could still maybe have better messages directing people to use that setting or modify their PATH
